From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Doe <john.doe@somewhere.on.planet>
Date: Mon, 28 Apr 2025 15:51:34 +0000
Subject: Patching kernel sunxi64 files
 arch/arm64/boot/dts/allwinner/overlay/Makefile
 arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-i2s3.dtso

Signed-off-by: John Doe <john.doe@somewhere.on.planet>
---
 arch/arm64/boot/dts/allwinner/overlay/Makefile              |  4 +-
 arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-i2s3.dtso | 85 ++++++++++
 2 files changed, 88 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/allwinner/overlay/Makefile b/arch/arm64/boot/dts/allwinner/overlay/Makefile
index 97bec2bf0615..bd70914bfbf8 100644
--- a/arch/arm64/boot/dts/allwinner/overlay/Makefile
+++ b/arch/arm64/boot/dts/allwinner/overlay/Makefile
@@ -79,11 +79,13 @@ dtb-$(CONFIG_ARCH_SUNXI) += \
 	sun50i-h616-light.dtbo \
 	sun50i-h616-i2c0-pi.dtbo \
 	sun50i-h616-i2c1-pi.dtbo \
 	sun50i-h616-i2c2-pi.dtbo \
 	sun50i-h616-gpu.dtbo \
-	sun50i-h616-usb0-host.dtbo
+	sun50i-h616-usb0-host.dtbo \
+	sun50i-h616-i2s3.dtbo
+
 
 scr-$(CONFIG_ARCH_SUNXI) += \
 	sun50i-a64-fixup.scr \
 	sun50i-h5-fixup.scr \
 	sun50i-h6-fixup.scr \
diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-i2s3.dtso b/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-i2s3.dtso
new file mode 100644
index 000000000000..6d5854a165b8
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-h616-i2s3.dtso
@@ -0,0 +1,85 @@
+/dts-v1/;
+/plugin/;
+
+&{/aliases} {
+	pio = "/soc/pinctrl@300b000";
+	ahub_daudio3_pins_a = "/soc/pinctrl@300b000/ahub_daudio30";
+	ahub_daudio3_pins_b = "/soc/pinctrl@300b000/ahub_daudio31";
+	ahub_daudio3_pins_c = "/soc/pinctrl@300b000/ahub_daudio32";
+	ahub_daudio3_pins_d = "/soc/pinctrl@300b000/ahub_daudio3_sleep";
+	};
+
+&pio {
+		ahub_daudio30: ahub_daudio30 {
+			pins = "PH5", "PH6", "PH7";
+			function = "i2s3";
+			drive-strength = <0x14>;
+			bias-disable;
+//			phandle = <0xb0>;
+						};
+		ahub_daudio31: ahub_daudio31 {
+			pins = "PH8";
+			function = "i2s3_dout0";
+			drive-strength = <0x14>;
+			bias-disable;
+//			phandle = <0xb1>;
+						};
+		ahub_daudio32: ahub_daudio32  {
+			pins = "PH9";
+			function = "i2s3_din0";
+			drive-strength = <0x14>;
+			bias-disable;
+//			phandle = <0xb2>;
+						};
+		ahub_daudio3_sleep: ahub_daudio3_sleep  {
+			pins = "PH5", "PH6", "PH7", "PH8", "PH9";
+			function = "gpio_in";
+			drive-strength = <0x14>;
+			bias-disable;
+//			phandle = <0xb3>;
+						};
+};
+
+&{/soc}  {
+	ahub3_plat: ahub3_plat {
+		#sound-dai-cells = <0>;
+		compatible	= "allwinner,sunxi-snd-plat-ahub";
+		apb_num		= <2>;	/* for dma port 5 */
+		dmas		= <&dma 5>, <&dma 5>;
+		dma-names	= "tx", "rx";
+		playback_cma	= <128>;
+		capture_cma	= <128>;
+		tx_fifo_size	= <128>;
+		rx_fifo_size	= <128>;
+
+		tdm_num		= <3>;
+		tx_pin		= <0>;
+		rx_pin		= <0>;
+		status		= "okay";
+
+		pinctrl-names = "default", "sleep";
+		pinctrl-0 = <&ahub_daudio30>, <&ahub_daudio31>;
+		pinctrl-1 = <&ahub_daudio3_sleep>;
+
+		};
+	ahub3_mach: ahub3_mach {
+		compatible = "allwinner,sunxi-snd-mach";
+		soundcard-mach,name = "ahubi2s3";
+		soundcard-mach,format		= "i2s";
+		soundcard-mach,frame-master	= <&ahub3_cpu>;
+		soundcard-mach,bitclock-master	= <&ahub3_cpu>;
+		/* soundcard-mach,frame-inversion; */
+		/* soundcard-mach,bitclock-inversion; */
+		soundcard-mach,slot-num		= <2>;
+		soundcard-mach,slot-width	= <32>;
+		status = "okay";
+		ahub3_cpu: soundcard-mach,cpu {
+			sound-dai = <&ahub3_plat>;
+			soundcard-mach,pll-fs	= <0x04>;
+			soundcard-mach,mclk-fs	= <0x200>;
+						};
+
+		ahub3_codec: soundcard-mach,codec {
+					};
+			};
+};
-- 
Created with Armbian build tools https://github.com/armbian/build

